src/libraries/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.Windows.cs

```diff
@@ -197,6 +197,7 @@ public async Task Connection_UnderDifferentUsers_CurrentUserOnlyOnServer_Invalid
                 {
                     // When CurrentUserOnly is only on client side and asks for ReadOnly access, the connection is not rejected
                     // but we get the UnauthorizedAccessException on the client regardless.
+                    serverTask.Wait(10_000);
```

Comment:
10_000 ms is 10 s, I know it's an OuterLoop tests but it seems too much. Perhaps we could start with 3 seconds?

Or just use `ManualResetEvent`?

```cs
using ManualResetEvent connected = new (false);
Task serverTask = server.WaitForConnectionAsync(cts.Token);

_testAccountImpersonator.RunImpersonated(() =>
{
    using (var client = new NamedPipeClientStream(".", name, clientDirection, clientPipeOptions))
    {
        Assert.Throws<UnauthorizedAccessException>(() => client.Connect());
        
        connected.Set();
    }
});

if (serverPipeOptions == PipeOptions.None && clientPipeOptions == PipeOptions.CurrentUserOnly && clientDirection == PipeDirection.In)
{
    // When CurrentUserOnly is only on client side and asks for ReadOnly access, the connection is not rejected
    // but we get the UnauthorizedAccessException on the client regardless.
    Assert.True(connected.WaitOne(TimeSpan.FromSeconds(5)));
    Assert.True(serverTask.IsCompletedSuccessfully)
```

Reply:
```c#
Assert.True(serverTask.Wait(TimeSpan.FromSeconds(10)));
```
?

Reply:
Can we keep this change minimal to quickly unblock the CI?

Reply:
@MihaZupan suggestion is better than mine, let's just use it.

```suggestion
                    Assert.True(serverTask.Wait(TimeSpan.FromSeconds(10)));
```

I am going to merge it right away to save us some time.